bindings/python/soletta/mainloop.pyx

```diff
@@ -0,0 +1,90 @@
+import asyncio
+from libc.stdint cimport uint32_t
+
+cdef extern from "sol-mainloop.h":
+	struct sol_mainloop_implementation:
+		void run()
+		void init()
+		void quit()
+		void * idle_add(bint (*)(void*), void*)
+		bint idle_del(void*)
+		void* timeout_add(uint32_t timeout_ms, bint (*cb)(void *), const void *)
+		bint timeout_del(void*)
+		void shutdown()
+		pass
+
+	bint sol_mainloop_set_implementation(const sol_mainloop_implementation *)
+
+
+cdef void wrap_sol_init():
+	pass
+
+cdef void wrap_sol_shutdown():
+	pass
+
+cdef void wrap_sol_quit():
+	asyncio.get_event_loop().stop()
+
+cdef void wrap_sol_run():
+	asyncio.get_event_loop().run_forever()
+
+cdef void *wrap_sol_timeout_add(uint32_t timeout_ms, bint (*cb)(void *), const void *data):
+	
+	@asyncio.coroutine
+	def task():
+		while cb(data):
+			yield from asyncio.sleep(timeout_ms / 1000)
```

Comment:
btw in C it is fine to call "sol_timeout_del()" (or sol_idle_del()) and return False in cb(data). Do you need to do something to account for that?

Test in C using your asyncio:

```c
struct sol_timeout *t;
bool cb(void *data) {
    sol_timeout_del(t);
    return false;    
}

    struct sol_timeout *t = sol_timeout_add(100, cb, NULL);
```